src/aggregate.d

```diff
@@ -639,6 +639,16 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
         return type;
     }
 
+    /****************************************
+     * Returns `true` if `this` aggregate has an `InputRange` interface
+     */
+    final bool isInputRange()
```

Comment:
As `foreach` already tests for ranges, I would have to guess this test exists somewhere in dmd already